abci/client/grpc_client.go

```diff
@@ -68,8 +68,8 @@ func (cli *grpcClient) OnStart() error {
 	go func() {
 		// Use a separate function to use defer for mutex unlocks (this handles panics)
 		callCb := func(reqres *ReqRes) {
-			cli.mtx.Lock()
```

Comment:
This should remain a write-lock, no?

Reply:
I can't see any data in the client it's writing/modifying.

Reply:
True, but I just don't know enough about the ABCI clients to be comfortable giving this change an ACK or not :-/

Reply:
cool, I unwound a bunch